task3_snkrs/spiders/snkrs_spider.py

```diff
@@ -0,0 +1,86 @@
+import scrapy
+from task3_snkrs.items import SnkrsItem
+
+
+class SnkrsSpider(scrapy.Spider):
+    name = 'snkrs'
+    response = None
+    start_urls = [
+        'https://www.snkrs.com/'
+    ]
+
+    def parse(self, response):
+        urls_s = response.css('ul.idparent-63 li a')
+        return [response.follow(url_s, callback=self.parse_pages) for url_s in urls_s]
+
+    def parse_pages(self, response):
+        yield from self.parse_product_urls(response)
+
+        yield from [response.follow(url_s, callback=self.parse_pages) for url_s in response.css('li.pagination_next a')]
+
+    def parse_product(self, response):
+        self.response = response
+        item = SnkrsItem()
+        item['item_id'] = self.get_item_id()
+        item['name'] = self.get_name()
+        item['brand'] = self.get_brand()
+        item['availability'] = self.get_availability()
+        item['currency'] = self.get_currency()
+        item['description'] = self.get_description()
+        item['image_urls'] = self.get_images_urls()
+        item['skus'] = self.get_skus()
+
+        return item
+
+    def parse_product_urls(self, response):
+        urls_s = response.css('a.product_img_link')
+        return [response.follow(url_s, callback=self.parse_product) for url_s in urls_s]
+
+    def get_item_id(self):
+        return self.response.css('.product_id::text').get()
```

Comment:
`self.response.css('.product_id::text').getall()[0]`
So there will be a visible errors if we are missing item name.